src/actions/index.js

```diff
@@ -376,6 +378,14 @@ export function removeCommunityModerator (community, moderatorId, prevProps) {
   }
 }
 
+export function removeCommunityMember (community, userId, cacheId, prevProps) {
+  return {
+    type: REMOVE_COMMUNITY_MEMBER,
+    payload: {api: true, path: `/noo/community/${community.id}/member/${userId}`, method: 'DELETE'},
+    meta: {slug: community.slug, userId, cacheId, prevProps}
```

Comment:
Yes, agreed prevProps and cacheId is fishy here.
Middleware idea is smart. I think it would have to save a diff, because other actions can potentially fire in between.
So for now should I just remove the optimistic stuff here?